File: frontend/replace_fonts.py
import os

directory = "src"

replacements = [
    ("text-[10px]", "text-[13px]"),
    ("text-[11px]", "text-[13px]"),
    ("text-[12px]", "text-[13px]"),
    ("text-xs", "text-[13px]")
]

def process_file(filepath):
    try:
        with open(filepath, "r", encoding="utf-8") as f:
            content = f.read()
        
        original = content
        for old, new in replacements:
            content = content.replace(old, new)
            
        if content != original:
            with open(filepath, "w", encoding="utf-8") as f:
                f.write(content)
            print(f"Updated {filepath}")
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith((".tsx", ".ts")):
            process_file(os.path.join(root, file))

File: frontend/replace_fonts.js
const fs = require('fs');
const path = require('path');

const directory = 'src';

const replacements = [
    { old: /text-\[10px\]/g, new: 'text-[15px]' },
    { old: /text-\[11px\]/g, new: 'text-[1px]' },
    { old: /text-\[12px\]/g, new: 'text-[1px]' },
    { old: /text-xs/g, new: 'text-[13px]' }
];

function processFile(filepath) {
    try {
        const content = fs.readFileSync(filepath, 'utf8');
        let newContent = content;

        for (const replacement of replacements) {
            newContent = newContent.replace(replacement.old, replacement.new);
        }

        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated ${filepath}`);
        }
    } catch (e) {
        console.error(`Error processing ${filepath}:`, e);
    }
}

function traverseDirectory(dir) {
    const files = fs.readdirSync(dir);

    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory()) {
            traverseDirectory(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
            processFile(fullPath);
        }
    }
}

traverseDirectory(directory);
